hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -595,41 +625,21 @@ data StepState tx = StepState
   , ledger :: Ledger tx
   }
 
--- | Asserts that the update function will update the state (return a NewState) for this Event
+-- | Retrieves the latest 'HeadState' from within 'runEvents'.
+getState :: (MonadState (StepState tx) m) => m (HeadState tx)
+getState = headState <$> get
+
+-- | Calls 'update' and 'aggregate' to drive the 'runEvents' monad forward.
 step ::
-  (MonadState (StepState tx) m, HasCallStack, IsChainState tx) =>
+  (MonadState (StepState tx) m, IsChainState tx) =>
   Event tx ->
-  m (HeadState tx)
+  m (Outcome tx)
```

Comment:
There was a refactoring introducing `getState` to precisely not have this return value and be able to not turn off the unused do binds warning.

Should: remove the need for this return value, have two variants or `void` invocations without the warning disabled

Reply:
We thought it is not that important since this is test code. We could also add the `Outcome` as part of the `StepState` and avoid these warnings.